libraries/stdlib/samples/test/samples/text/strings.kt

```diff
@@ -508,6 +508,104 @@ class Strings {
         assertPrints(mixedColor, "brown&blue")
     }
 
+    @Sample
+    fun toPattern() {
+        val string = "this is a regex"
+        val pattern = string.toPattern(1)
+        assertPrints(pattern.pattern(), string)
+        assertTrue(pattern.flags() == 1)
+    }
+
+    @Sample
+    fun encodeToByteArray() {
```

Comment:
It would be nice to also include an example illustrating:
```
Any malformed char sequence is replaced by the replacement byte sequence.
```

Although, it could be problematic given a slightly different behavior on JVM and Native.